app/controllers/album.js

```diff
@@ -59,4 +59,24 @@ exports.purchase = (req, res, next) => {
 
   });
 
+};
+
+exports.purchaseList = (req, res, next) => {
+
+  try {
+    validationResult(req).throw();
+  } catch (err) {
+    return next(error.invalidUserId);
+  }
+
+  id = req.query.id;
+
+  if((id != req.user.id) && !req.user.isAdmin){
+    return next(error.notAnAdmin);
+  }else{
```

Comment:
The same I told you in another PR, if you have something like 
```
if(...) {
  return ...
} else {
  more code..
}
``` 
you can replace it with:
```
if(...) {
  return ...
}
more code..
```